### ISSUE TITLE
Different values of "memory stats" results depending on access method: redis-cli vs redis-py

### ISSUE THREAD

**Issue by chill1n** [andymccurdy/redis-py#1443]
**Version**: What redis-py and what redis version is the issue happening on?
redis-py 3.5.3
redis 6.0.8

**Platform**: What platform / version? (For example Python 3.5.1 on Windows 7 / Ubuntu 15.10 / Azure)
Python 3.8.5 on Ubuntu 20.04

**Description**: Description of your issue, stack traces from errors and code that reproduces the issue

I've noticed substantially different values in the output of "memory stats" obtained via two different methods: redis-cli vs redis-py.
For example (after "flushall"):
1. redis-cli: 
    23) "dataset.bytes"
    24) (integer) 46694
2. redis-py:
    dataset.bytes   129550 

I expect to see two values that are identical for dataset.bytes, but these are consistently different (although static for each method). I am assuming that redis-cli gives the correct output. How can the difference be explained?

**Comment by andymccurdy** [comment]
I don't see how such a discrepancy is possible. Are you certain you're connecting both clients to the same Redis instance?

**Comment by chill1n** [comment]
Yes, I double checked.

You can check with this one-liner:
import redis; r = redis.Redis(host="localhost", port=6379, db=0); ms = r.memory_stats()['dataset.bytes']; print(ms)

Just observed that the "dataset.bytes" value drops after I exit the python CLI. This suggests that this parameter depends on the interface method. Maybe it's just my (mis)understanding of the dataset.bytes parameter?
So, it seems to be a "feature" of redis. I think you can close this issue although I would still appreciate if someone can tell me how to measure the amount of (current) data memory that's being used by redis.

**Comment by andymccurdy** [comment]
It appears that each connection adds some amount of overhead to `dataset.bytes` (which after reading the MEMORY STATS docs here: https://redis.io/commands/memory-stats seems like a possible server bug). I connected one client via redis-cli, and `MEMORY STATS` consistently reported the same `dataset.bytes` value. As soon as I connected a second client via redis-cli, the `dataset.bytes` value went up on both clients (both were the same). When I disconnected one client, the `dataset.bytes` value dropped.

Closing this as it doesn't appear to be a redis-py issue. I'd suggest asking for more information on the Redis google group mailing list.
